lib/std/x/net/tcp.zig

```diff
@@ -0,0 +1,388 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2021 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+
+const std = @import("../../std.zig");
+
+const os = std.os;
+const fmt = std.fmt;
+const mem = std.mem;
+const builtin = std.builtin;
+const testing = std.testing;
+
+const IPv4 = std.x.os.IPv4;
+const IPv6 = std.x.os.IPv6;
+const Socket = std.x.os.Socket;
+
+/// A generic TCP socket abstraction.
+const tcp = @This();
+
+/// A union of all eligible types of socket addresses over TCP.
+pub const Address = union(enum) {
```

Comment:
Seems weird that the Address struct is in a module named `tcp` when it can also be used for `udp` sockets or any other type of socket that leverages the IP protocol "InternetProtocol".

Reply:
How will the UDP Address type differ from the TCP address type?

Reply:
An IP address is orthogonal to UDP/TCP.  It' in a different layer of the packet stack.  There is no difference between an IP address used for a UDP packet and a TCP packet.  Having a different type for each one would be like having a different `File` type in the standard library for `.c` files and `.zig` files.

Reply:
Oh, my bad I misread your comment. Having a `IP.Address` rather than individual `TCP.Address` and `UDP.Address` modules makes sense. I'll make the change tomorrow; thanks!

Reply:
Pushed a new commit that moves `tcp.Address` to `ip.Address` with `ip` being a new module.